### PR TITLE
Add Browser#openAndWait for convenience(Fixes #28)

### LINKED ISSUE #28: Add Browser#openAndWait for convenience
It is very common to do:

```java
browser.open(someViewUrl)
    .waitUpTo(DEFAULT_DURATION);
```

So we could shorten this to:

```java
browser.openAndWait(someViewUrl, DEFAULT_DURATION)
```

We could even add an overload that did not accept a duration, and used some (configurable? eventually) default.

### PR DESCRIPTION
**Pull request by abocz**
I'm not sure what the best way to make the predetermined duration for `openAndWait(ViewUrl<T> viewUrl` configurable.  Also not sure if you wanted to add javadocs here, but I gave it a shot.

### CODE REVIEW

**Comment by alechenninger** [comment]
I don't like the 2-minute baked in wait time. I would just add the duration as an argument. You could have two overrides that follow the similar pattern as `waitUpTo` does as well (one accepts duration, other accepts integer and chrono unit).

What we really probably want is this to be implemented: https://github.com/darcy-framework/synq/issues/17

Then you could just `browser.open(urlFor.myView()).await();`

File: src/main/java/com/redhat/darcy/web/api/Browser.java [hunk 1]
```diff
@@ -68,6 +72,37 @@
...
+     * open the url. Will block the thread for a maximum of the two minutes, at which point a
+     * {@link com.redhat.synq.TimeoutException} will be thrown.
+     * @param viewUrl If you don't have a {@link com.redhat.darcy.web.api.ViewUrl} instance, but you
+     * know the url and the resulting {@link com.redhat.darcy.ui.api.View}, see
+     * {@link #open(String, com.redhat.darcy.ui.api.View)}.
+     * @return The awaited view once it has met all criteria for loading
+     */
+    default <T extends View> T openAndWait(ViewUrl<T> viewUrl) {
+        return open(viewUrl).waitUpTo(Duration.of(2, ChronoUnit.MINUTES));
+    }
```
Comment (alechenninger) [inline src/main/java/com/redhat/darcy/web/api/Browser.java]: If you remove this override I will merge this -- I don't like the hard coded 2 minute wait time here

File: src/main/java/com/redhat/darcy/web/api/Browser.java [hunk 2]
```diff
@@ -68,6 +72,37 @@
...
+     * {@link com.redhat.synq.TimeoutException} will be thrown.
+     * @param viewUrl If you don't have a {@link com.redhat.darcy.web.api.ViewUrl} instance, but you
+     * know the url and the resulting {@link com.redhat.darcy.ui.api.View}, see
+     * {@link #open(String, com.redhat.darcy.ui.api.View)}.
+     * @param duration Maximum specified duration of the view loading
+     * @return The awaited view once it has met all criteria for loading
+     */
+    default <T extends View> T openAndWait(ViewUrl<T> viewUrl, Duration duration) {
+        return open(viewUrl).waitUpTo(duration);
+    }
```
Comment (alechenninger) [inline src/main/java/com/redhat/darcy/web/api/Browser.java]: I would also add an override that accepts a `Long` and a `ChronoUnit`

File: src/main/java/com/redhat/darcy/web/api/Browser.java [hunk 3]
```diff
@@ -68,6 +72,37 @@
     <T extends View> Event<T> open(String url, T destination);
 
     /**
+     * Constructs an {@link com.redhat.synq.Event} that will opens the URL and block until the
+     * associated {@link com.redhat.darcy.ui.api.View} is loaded, as defined by the {@link ViewUrl}
+     * parameter, and then calls {@link Event#waitUpTo(java.time.Duration)} to open the url. Will
+     * block the thread for a maximum of the specified duration, at which point a
+     * {@link com.redhat.synq.TimeoutException} will be thrown.
```
Comment (alechenninger) [inline src/main/java/com/redhat/darcy/web/api/Browser.java]: I think that an `Event` is constructed is kind of implementation detail for this API, and not the first line I would put here. What would be most informative is simply that it opens the url and immediately starts waiting for the view to be loaded. The other API's start with this because that's all they do -- construct an `Event`, and you that's important because if you don't wait for it, the page will never open. That fact that this is different is kind of buried in this java doc. Of course, the name of the method does make it more clear, but no reason the javadoc shouldn't be clear about it too.

File: src/main/java/com/redhat/darcy/web/api/Browser.java [hunk 4]
```diff
@@ -68,6 +72,37 @@
...
+     * @param viewUrl If you don't have a {@link com.redhat.darcy.web.api.ViewUrl} instance, but you
+     * know the url and the resulting {@link com.redhat.darcy.ui.api.View}, see
+     * {@link #open(String, com.redhat.darcy.ui.api.View)}.
+     * @param duration Maximum specified duration of the view loading
+     * @return The awaited view once it has met all criteria for loading
+     */
+    default <T extends View> T openAndWait(ViewUrl<T> viewUrl, Duration duration) {
```
Comment (alechenninger) [inline src/main/java/com/redhat/darcy/web/api/Browser.java]: Would `openAndWaitUpTo` be too verbose?

File: src/main/java/com/redhat/darcy/web/api/Browser.java (hunk 3 above)
Comment (abocz) [inline src/main/java/com/redhat/darcy/web/api/Browser.java]: Thanks for these comments, I revised the javadocs accordingly.

File: src/main/java/com/redhat/darcy/web/api/Browser.java (hunk 2 above)
Comment (abocz) [inline src/main/java/com/redhat/darcy/web/api/Browser.java]: Done.

File: src/main/java/com/redhat/darcy/web/api/Browser.java (hunk 4 above)
Comment (abocz) [inline src/main/java/com/redhat/darcy/web/api/Browser.java]: > Would openAndWaitUpTo be too verbose?

nah!

**Comment by alechenninger** [comment]
Awesome, thanks!
